homeassistant/components/notify/slack.py

```diff
@@ -22,6 +23,8 @@
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
     vol.Required(CONF_API_KEY): cv.string,
     vol.Required(CONF_CHANNEL): cv.string,
+    vol.Optional(CONF_USERNAME, default=None): cv.string,
+    vol.Optional(CONF_ICON, default=None): cv.string,
```

Comment:
That's odd because I see in cv.string it *should* do like you say, but in my testing it never generated an error.  Easy enough fix to do it the correct way tho.